### ISSUE TITLE
[Compose] Image is not load

### ISSUE THREAD

**Issue by Patresko** [coil-kt/coil#826]
After updating to latest version, followed by docs for migration from Accompanist, images are not visible at all.

`@Composable
fun LiftEntry(
    lift: LanguageLift,
    liftData: LiftData,
    navController: NavController
) {
    Row(
        horizontalArrangement = Arrangement.SpaceBetween,
        modifier = Modifier
            .clickable {
                navController.currentBackStackEntry?.arguments =
                    Bundle().apply {
                        putParcelable("lift", liftData)
                    }
                navController.navigate("lift_detail_screen")
            }
            .fillMaxWidth()
            .padding(vertical = 10.dp)
    ) {
        val url = "$IMG_URL${lift.photoId}&size=mediumJpg"

        // Advanced
        Image(
            painter = rememberImagePainter(
                data = "https://shop.dhollandia.com/APP?function=photo&id=Da0hSctZrOVjGw&size=mediumJpg",
                builder = {
                    placeholder(R.drawable.no_image_available)
                }
            ),
            contentDescription = null,
            modifier = Modifier.widthIn(min = 100.dp).
                    fillMaxWidth(0.3f)
                .clip(RoundedCornerShape(10.dp)),
            contentScale = ContentScale.FillWidth
        )

        Column(
            horizontalAlignment = Alignment.Start,
            verticalArrangement = Arrangement.Top,
            modifier = Modifier
                .padding(horizontal = 10.dp)
                .fillMaxWidth(0.9f)
        ) {
            Text(text = lift.code, color = MaterialTheme.colors.primaryVariant, fontSize = 28.sp)
            Text(text = lift.title, color = Color.White)
        }

        Icon(
            Icons.Filled.ArrowForwardIos,
            contentDescription = "Next",
            modifier = Modifier
                .size(26.dp)
                .fillMaxWidth(),
            tint = MaterialTheme.colors.primaryVariant
        )
    }
}`

If you remove all modifier from Image and keep only .size(120.dp) image is visible. 

Seems that problem is only if Image is placed in Row. In row, picture does not accepting any modifier as .FillMaxWidth(), FillMaxSize() and so on and image does not show up on the screen. Only way how to show image inside the row is set .size modifier on the image.

**Comment by rovkinmax** [comment]
Image loader does not start request to load image, because `size` on this line https://github.com/coil-kt/coil/blob/main/coil-compose-base/src/main/java/coil/compose/ImagePainter.kt#L174 is always zero

## To Reproduce
This sample not show image. Only if set up defaultMinSize

```
    Box(
            modifier = Modifier.fillMaxSize(),
            contentAlignment = Alignment.BottomCenter,
    ) {

        Column(modifier = Modifier
                .fillMaxSize()
                .verticalScroll(rememberScrollState())) {

            (0..4).forEach {
                Text(text = "$it item", modifier = Modifier.padding(all = 20.dp))
            }


            Image(
                    painter = rememberImagePainter(
                            data = "https://cdn.pixabay.com/photo/2015/04/23/22/00/tree-736885__480.jpg",
                            imageLoader = LocalImageLoader.current,
                    ),
                    contentDescription = null,
                    contentScale = ContentScale.FillWidth,
                    modifier = Modifier
                            .fillMaxWidth()
                            //.size(300.dp)
                            .padding(16.dp)
                            .clickable { },
            )

        }
    }
```

**Comment by colinrtwhite** [comment]
This might be intended behaviour, but I'll look a bit deeper. The Compose framework needs to call `ImagePainter.draw`. If the `Image` doesn't have a resolved size then it won't be drawn. That said, I'd expect `fillMaxWidth` to give the `Image` a resolved size. Can you verify if `ImagePainter.draw` is called?

**Comment by rovkinmax** [comment]
@colinrtwhite If you about this method of ImagePainter
```
    override fun DrawScope.onDraw() {
        // Update the draw scope's current size.
        drawSize = size

        // Draw the current painter.
        painter?.apply { draw(size, alpha, colorFilter) }
    }
```
Then yes, It called many times with non zero size for image from the sample above

**Comment by colinrtwhite** [comment]
@rovkinmax I tried out your sample code and `ImagePainter.draw` isn't called. This is intended behaviour as `Column` needs a height to figure out how tall the `Image` should be. Setting `size(300.dp)` or `height(300.dp)` renders the image correctly.

@Patresko I tried your code sample and an image with `fillMaxWidth` is working as intended. I added a test for it [here](https://github.com/coil-kt/coil/pull/846).
